### PR TITLE
Improvements to Settings Page UI, Backend Integration, Alertmanager Addition

### PR DESCRIPTION
**Pull request by lwu15**
- Refactored the Settings page UI and integrated it with the backend using routers. Applied styling to the Settings page.
- Added Alertmanager to Docker Compose and updated Prometheus configuration.
- Updated Docker Compose to mount Alertmanager and successfully send alerts.
- Removed the URL from alertmanager.yml and stored it separately.

Testing and Configuration: Code has been locally tested with docker compose up for altering. When testing, be sure to update the Slack API URL in the alertmanager.yml file with the actual URL before running the docker compose up command. (Note that the URL is invalidated once included in a commit). Then go to the localhost for prometheus and alertmanager respectively. 

Also, successfully tested that alert preferences are stored in the database.

### CODE REVIEW

File: src/pages/Settings.jsx [hunk 1]
```diff
@@ -1,128 +1,256 @@
...
+    if (
+      !alertPreferences.Email &&
+      !alertPreferences.Slack &&
+      !alertPreferences.InApp
+    ) {
+      setSettingsFeedback('Select at least one alert preference.');
+      return;
+    }
+
+    if (!password.trim()) {
```
Comment (echochs1) [inline src/pages/Settings.jsx:92]: Doesn't check entered password with hashed password in DB

Comment (lwu15) [reply src/pages/Settings.jsx:92]: updated

**Comment by lwu15** [comment]
@echochs1  Elliot, I appreciate your code review. Following our discussion, I've made the following changes:

- Removed `JSON.stringify` for password comparison since it was duplicative. Also, I'm now verifying users on the backend/added password verification for updating settings.
- - Planning to streamline the Settings page UI for a better user experience.

Additionally, updated `authControllers` to improve error propagation. Previously, it was not using `next` for the error messages/sending them directly but now correctly sends the error messages for the login page.

I've also commented out unnecessary `console.log` statements.

Ready for your review and feedback.

**Review by echochs1 — APPROVED** [review]
Looks great!

**Pull request merged**